src/command.ts

```diff
@@ -27,22 +31,54 @@ export class Command {
             },
         };
 
-        const exitCode = await exec.exec(executable, args, finalExecOptions);
+        if (execOptions.group) {
+            const groupName = [ executable, ...args ].join(" ");
+            core.startGroup(groupName);
+        }
+
+        try {
+            const exitCode = await exec.exec(executable, args, finalExecOptions);
 
-        if (execOptions.ignoreReturnCode !== true && exitCode !== 0) {
-            // Throwing the stderr as part of the Error makes the stderr show up in the action outline,
-            // which saves some clicking when debugging.
-            let error = `${path.basename(executable)} exited with code ${exitCode}`;
-            if (stderr) {
-                error += `\n${stderr}`;
+            if (execOptions.ignoreReturnCode !== true && exitCode !== 0) {
+                // Throwing the stderr as part of the Error makes the stderr show up in the action outline,
+                // which saves some clicking when debugging.
+                let error = `${path.basename(executable)} exited with code ${exitCode}`;
+                if (stderr) {
+                    error += `\n${stderr}`;
+                }
+                throw new Error(error);
             }
-            throw new Error(error);
+
+            return {
+                exitCode, output: stdout, error: stderr,
+            };
         }
 
-        return {
-            exitCode,
-            stdout,
-            stderr,
-        };
+        finally {
+            if (execOptions.group) {
+                core.endGroup();
+            }
+        }
+    }
+
+    public static async tag(image: string, tags: string[]): Promise<void> {
+        let dockerPath: string;
+
+        try {
+            // get docker cli
+            dockerPath = await io.which("docker", true);
+        }
+        catch (error) {
+            core.debug(error);
```

Comment:
it is probably better to just include this in the error message below.

`X ${err.message}. Make sure that "docker" is installed and on the PATH before running this action.`

Reply:
actually, this library is documented wrong.

It says:
```
If you check and the tool does not exist, it will throw.
```

but in reality, it returns `""`

So we are going to have to handle this differently by checking if dockerPath is not the empty string.

Reply:
Ahh Okay, we did this way in buildah action.

Reply:
[ /src/redhat-actions/podman-login ] 04 (main) $ node
Welcome to Node.js v14.16.0.
Type ".help" for more information.
> const io = require("@actions/io")
undefined
> io.which("asdf").then((whichResult) => console.log(`which result: "${whichResult}"`)).catch((err) => console.error(`which error: ${err}`))
Promise { <pending> }
> which result: ""
```

Reply:
ah, it throws if the second parameter, ambiguously named `check`, is true.

Reply:
So you can either set that parameter or you can handle the empty string case.

We likely have this issue in at least a few of our other actions, then.

Reply:
I remember any of our action using `check` thing

Reply:
this is what the error message looks like when it throws:

```
> io.which("asdf", true).then((whichResult) => console.log(`which result: "${whichResult}"`)).catch((err) => console.error(`which error: ${err}`))
Promise { <pending> }
> which error: Error: Unable to locate executable file: asdf. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

Reply:
yeah, we should add `check=true` to buildah too. I think PTR also does this for podman

Reply:
Yes, their message is good.

Reply:
Alternatively, `oc-login` https://github.com/redhat-actions/oc-login/blob/main/src/oc.ts just doesn't even put the absolute path, it just invokes `oc` and the shell finds it. no `which` required.

Reply:
This is a good way.